custom_components/test/fuzzing/fuzzer_utils/ValuePoolFuzzer.py

```diff
@@ -5,6 +5,117 @@
 from custom_components.test.fuzzing.fuzzer_utils.Fuzzer import Fuzzer
 from custom_components.test.fuzzing.fuzzer_utils.ValuePool import ValuePool
 
+def _generate_ranking(lists):
+    """
+    Generates ranking based on length of the lists.
+
+    :param lists: A list containing inner lists.
+    :type lists: list of lists
+    
+    :return: A ranking of the lists based on their length.
+    :rtype: list
+    """
+    rank = sorted(range(len(lists)), key=lambda x: (-len(lists[x]), x))
+    return rank
+
+def _get_repeater(lists, param_combi, m):
+    """
+    Calculates how often a list must be repeated if param_combi is greater than 1.
+
+    :param lists: A list containing inner lists.
+    :type lists: list of lists
+    :param param_combi: Number of parameters to be combined.
+    :type param_combi: int
+    :param m: The current index in the lists.
+    :type m: int
+    
+    :return: The number of times a list must be repeated.
+    :rtype: int
+    """
+    repeater = 1
+    param_combi_count = 1
+    while param_combi_count < param_combi:
+        if m >= param_combi_count:
+            repeater = repeater * len(lists[m-1])
+        param_combi_count += 1
+        
+    return repeater-1
+    
+def _generate_new_list(lists, param_combi, rank):
```

Comment:
Why is this function outside the `ValuePoolFuzzer` class? I think it is a private member of the `ValuePoolFuzzer` class.